File: Frontend/GymTrack/src/app/models/training.ts
import { ExerciseType } from "./exercise-type";

export interface Training {
    id: number;
    userId: number;
    exerciseTypeId: number;
    trainingDate: Date;
    duration: number;
    caloriesBurned: number;
    workoutIntensity: number;
    fatigue: number;
    notes?: string | null | undefined;
    exerciseType?: ExerciseType;
}

File: Backend/GymTrack/Entities/Training.cs
using System;

namespace GymTrack.Entities;

public class Training
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public int ExerciseTypeId { get; set; }
    public DateTime TrainingDate { get; set; }
    public int Duration { get; set; }
    public int CaloriesBurned { get; set; }
    public int WorkoutIntensity { get; set; }
    public int Fatigue { get; set; }
    public String? Notes { get; set; }

    public ExerciseType? ExerciseType { get; set; }
}
